File: src/main/java/org/goafabric/catalog/controller/ChargeItemController.java
package org.goafabric.catalog.controller;

import org.goafabric.catalog.job.chargeitem.ChargeItemRepository;
import org.goafabric.catalog.job.chargeitem.ChargeItemEo;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping(value = "chargeitems", produces = MediaType.APPLICATION_JSON_VALUE)
public class ChargeItemController {
    private final ChargeItemRepository repository;

    public ChargeItemController(ChargeItemRepository repository) {
        this.repository = repository;
    }

    @GetMapping("/{id}")
    public ChargeItemEo getById(@PathVariable String id) {
        return repository.findById(id).get();
    }

    @GetMapping("/findByCode")
    public ChargeItemEo findByCode(@RequestParam("code") String code) {
        return repository.findByCode(code);
    }

    @GetMapping("/findByDisplay")
    public List<ChargeItemEo> findByDisplay(@RequestParam("display") String display) {
        return repository.findByDisplayStartsWithIgnoreCase(display);
    }
}

File: src/main/kotlin/org/goafabric/catalog/controller/ChargeItemController.kt
package org.goafabric.catalog.controller

import org.goafabric.catalog.job.chargeitem.ChargeItemEo
import org.goafabric.catalog.job.chargeitem.ChargeItemRepository
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = ["chargeitems"], produces = [MediaType.APPLICATION_JSON_VALUE])
class ChargeItemController(private val repository: ChargeItemRepository) {
    @GetMapping("/{id}")
    fun getById(@PathVariable id: String): ChargeItemEo {
        return repository.findById(id).orElseThrow()!!
    }

    @GetMapping("/findByCode")
    fun findByCode(@RequestParam("code") code: String): ChargeItemEo? {
        return repository.findByCode(code)
    }

    @GetMapping("/findByDisplay")
    fun findByDisplay(@RequestParam("display") display: String): MutableList<ChargeItemEo> {
        return repository.findByDisplayStartsWith(display)
    }
}
